components/remote_settings/src/client.rs

```diff
@@ -18,6 +19,205 @@ const HEADER_BACKOFF: &str = "Backoff";
 const HEADER_ETAG: &str = "ETag";
 const HEADER_RETRY_AFTER: &str = "Retry-After";
 
+/// Internal Remote settings client API
+pub struct RemoteSettingsClient {
+    // This is immutable, so it can be outside the mutex
+    collection_name: String,
+    inner: Mutex<RemoteSettingsClientInner>,
+}
+
+struct RemoteSettingsClientInner {
+    storage: Storage,
```

Comment:
I believe that `Storage` will want to use `&mut` in a lot of its methods, since that's what `rusqlite::Connection::transaction` uses and in general we want exclusive access to the connection.  But I'm not totally sure this is right and maybe we should move it outside the mutex.